### PR TITLE
Refactor(`Person`): Change `StudyYear` attribute type from `String` to `StudyYear`

### PR DESCRIPTION
**Pull request by TeeJinSeng**

### CODE REVIEW

File: src/test/java/seedu/address/model/person/StudyYearTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
```
Comment (mackerelsoup) [inline src/test/java/seedu/address/model/person/StudyYearTest.java]: Retest these test classes with the most recent updated code

File: src/main/java/seedu/address/model/person/StudyYear.java
Comment (mackerelsoup) [inline src/main/java/seedu/address/model/person/StudyYear.java]: Duplicate with existing StudyYear class

File: src/test/java/seedu/address/model/person/StudyYearTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        assertFalse(StudyYear.isValidStudyYear("")); // empty string
+        assertFalse(StudyYear.isValidStudyYear(" ")); // spaces only
+        assertFalse(StudyYear.isValidStudyYear("SENIOR1")); // invalid academic level
+        assertFalse(StudyYear.isValidStudyYear("pri1")); // lower case
+        assertFalse(StudyYear.isValidStudyYear("SEC")); // no number provided
+        assertFalse(StudyYear.isValidStudyYear("JC 2")); // spaces in between
+        assertFalse(StudyYear.isValidStudyYear("POLY7")); // number out of range
+
+        // valid studyYears
+        assertTrue(StudyYear.isValidStudyYear("PRI6"));
```
Comment (ZavierCSJ) [inline src/test/java/seedu/address/model/person/StudyYearTest.java]: Nice variety of invalid formats. Can consider adding a test that asserts the full expected set or boundary values (e.g., PRI1, PRI6, SEC1, SEC5,
